test/integration/tcp_tunneling_integration_test.cc

```diff
@@ -606,6 +606,212 @@ TEST_P(TcpTunnelingIntegrationTest, BasicHeaderEvaluationTunnelingConfig) {
   closeConnection(fake_upstream_connection_);
 }
 
+// Verify the header evaluator is reused among multiple tcp requests on the same filter chain.
```

Comment:
does this test fail prior to this PR?  I thought it was largely a perf optimization so wouldn't affect headers on either end.

Reply:
I was too sure about the evaluator state. Read again and I find the evaluateHeaders is a const function and is naturally reusable.

This test case has very limited value and I am deleting it.